app/controllers/spree/billing/payments_controller.rb

```
@@ -40,6 +40,22 @@ def show
         @payment = @object
       end
 
+      def fire
+        return unless @payment.payment_source
```

if the condition is met, then this will be silently failed. You should raise and handle an exception in the **rescue_from** block